### DISCUSSION TITLE
Thermal Storage

### DISCUSSION THREAD

**Discussion by vincevomhaigst** [oemof/tespy#409] [General]
Hello everyone! Is there a possibility to implement a thermal storage in a Tespy design simulation? 
More detailed: A warmwater storage (tank) which is heated by a heat pump (connected with the condenser) and in the storage tank there is another heatexchanger for the "source and sink consumer side".

**Comment by fwitte** [comment]
Hi @vincevomhaigst,

sorry for coming to this a little bit late, I simply overlooked it.

Similar to the compressed air energy storage https://github.com/oemof/tespy/discussions/411, you'd have to consider a thermal storage like a source and a sink (hot water flows into the source side and the (remiaining) cold water out of the sink side). A coupled storage simulator would then determine the changes in temepratures for the outflow depending on the temperature of the inflow and its massflow (and storage losses etc.). Or, you could consinder a storage with a heat exchanger. For that you could use a simple heat exchanger and make the change of temperature of the fluid charging or discharging the storage a function of the storage's temperature.

best
